moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java

```diff
@@ -682,6 +670,65 @@ public void setupVersionHeader() {
         restClient.setHeader("Version", authenticator.getApiMajorVersion());
     }
 
+    private void setPersistentV3AuthHeaders() {
+        restClient.setHeader(SESSION_TTL, "120"); // 2h
+        restClient.setHeader(PREFER, "persistent-auth, csrf-protection");
+    }
+
+    private void resetClientSettings() {
+        restClient.setHeader(SESSION_TTL, "");
+        restClient.setHeader(PREFER, "");
+        restClient.setCookie(JSESSIONID, "");
+        restClient.setAuthentication(new HttpAuthentication() {
+            @Override
+            public String getHeaderValue() {
+                // empty authentication - e.g. not the basic one
+                return "";
+            }
+        });
+    }
+
+    private void updateClientBeforeCall() {
+        restClient.setHeader(FILTER, Boolean.toString(!authenticator.hasAdminPermissions()));
+        requestFactory.setCertificateHandlingMode(authenticator.getCertHandlingStrategy());
+        restClient.setRootUrl(authenticator.getApiUrl());
+    }
+
+    /**
+     * @param username username
+     * @param password password
+     * @return auth token depending on API version, if enforceBasicAuth is true then returns empty string
+     */
+    public String login(String username, String password) {
+        resetClientSettings();
+        updateClientBeforeCall();
+        restClient.setHttpBasicAuth(username, password);
+
+        Api api = restClient.loginV3();
```

Comment:
@jhernand hi Juan, we are trying to make a detection of the API version here in order to know if we should work with v3 or v4. But in order to ask the api what version it is, we need to log in the old way (basic auth). Than, once we are logged in, we can find out we are working with the new API and login again using the oauth.
Im not sure this is the intended use of the API. Is there a better way? Or, more importantly, will this even work after the v3 will be removed? :)